index.html

```diff
@@ -2582,179 +2582,141 @@ <h2>CBOR</h2>
-        </p>
+A DID document encoded in CBOR must be represented as a CBOR Map (major type 5) conforming to [[RFC7049]]. 
+Similar to the JSON encoding of a DID document, all top-level properties of the DID document MUST be represented by using the 
+property name as the name of the member of the CBOR map.  The values of properties of the data model described in <a href="#core-properties"> § 5. Core Properties</a>, including 
+all extensions, MUST be encoded in CBOR [[RFC7049]] by mapping property values to CBOR types as follows: </p>
+
+<ul>
+ <li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Numeric</a> values representable as IEEE754 MUST be represented as a Number type (major type 0 or 1).</li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Boolean</a> values MUST be represented as a Boolean literal (primitive type 21 for true and primitive type 20 for false).  </li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Sequence</a> value MUST be represented as an Array type (major type 4) .</li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Unordered sets</a> of values MUST be represented as an Array type (major type 4).</li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Sets of properties</a> MUST be represented as an map type (major type 5).</li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Empty values</a> MUST be represented as a null literal (primitive type 22). </li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Other values</a> MUST be represented as a String type. (major type 5)</li>
+<li><a href="https://tools.ietf.org/id/draft-ietf-cbor-7049bis-12.html#section-3.2.2">Indefinite-length items</a> MUST be made definite legth.</li>
+<li><a href="https://www.rfc-editor.org/rfc/rfc3339.txt">Datetime </a> MUST be encoded in UTF-8 (major type 3). This datetime value MUST be normalized to UTC 00:00, as indicated by the trailing "Z".</li>
+</ul>
+
+
+<p>Semantic interoperability with other core representations of the DID document is possible with additional constaints 
+  to the CBOR representation.  Additionally, for lossless conversion between other core representations a deterministic 
+  representation is desirable when producing DID Documents that are represented as CBOR, in addition to the suggestions
+  in section 3.9 of the CBOR [[RFC7049]] specification for deterministic mappings, 
+  the following constraints of the DID Document model MUST be followed:
+  These additional constaints include: </p>
+
+<ul>
+    <li>Property names MUST be represented as a byte string array (major type 2) and contain only UTF-8 strings. </li>
+    <li>All member properties of a CBOR encoded DID document MUST be either represented in the `@context`, be present in the 
```

Comment:
Why would a CBOR representation have a `@context`? Is this the same as CBOR-LD?

Reply:
@peacekeeper  

Totally valid to have `@context` in CBOR representation.   That is why I included: 

```
The property name @context MAY be used in the CBOR map and 
has the same semantic meaning as JSON-LD producers of a DID document in JSON representation.
```

 As  CBOR can support all JSON data types (including JSON-LD) for translation between the DID document model as the JSON generic data model is just a subset of the CBOR generic data model as such including the `@context`, which  gives semantics to the properties, but doesn't force json-LD processing.  My understanding of CBOR-LD is externally dictionary by reference using the `@context` in combination with an algorithm for iterating over properties and will require proper declaration to give the appropriate syntax for CBOR-LD `processing` required, something like `content-type = application/cbor+ld`.   We can do this, I think, with the current wording, but will eventually need to add that subsection on how that includes the appropriate CBOR tags and algorithm, etc.

Reply:
Okay I understand that all JSON including JSON-LD can be directly mapped to CBOR. But if we define a CBOR representation of the DID document ADM, then the production rules should clearly say if a `@context` gets added or not.

- In the JSON-LD production rules, the addition of `@context` is clearly defined.
- In the JSON production rules, there is support for removing all mention of `@context` (https://github.com/w3c/did-core/pull/396).

The CBOR production rules should also make it clear whether or not `@context` gets added.

Reply:
@peacekeeper why? should every section specifically state if properties related to processing directives are included or excluded?

Seems better to just say what is required, and how preserve by default works.

In a world where there are 10 representations, the approach of having each section comment about the other sections will not work.

Reply:
Agree with @OR13 -- what is being suggested by @peacekeeper is not a scalable approach; it would require representations to know about other representations, which is strange.

Reply:
well, the ADM using JSON-LD is strange to me as evident by the `@type` issue that we are currently having.    @peacekeeper,  I think that from the discussion regarding ADM that we agreed that `@context` is optional in the JSON representations. It is only when you combine it with MIME = `application/did+json+ld that you explicitly state that JSON-LD processing is required.

Reply:
Technically JSON-LD processing is still not required with `application/did+json+ld` per the spec language today.

Reply:
interesting, thanks for the clarification!